qemu/tests/fio_perf.py

```diff
@@ -0,0 +1,224 @@
+        if len(re.findall("Disk %s" % num, disks_str)) >= 1:
+            disks = re.findall("Disk %s.*\n" % num, disks_str)
+            break
+    return disks
+
+
+def get_version(session, result_file, guest_ver_cmd, type, driver_format, timeout):
+    kvm_ver = utils.system_output("rpm -qa qemu-kvm-rhev && rpm -qa qemu-kvm")
+    host_ver = os.uname()[2]
+
+    result_file.write("### kvm-userspace-ver : %s\n" % kvm_ver)
+    result_file.write("### kvm_version : %s\n" % host_ver)
+
+    if driver_format != "ide":
+        result = session.cmd_output(guest_ver_cmd, timeout)
+        if type == "windows":
+            guest_ver = re.findall(".*?(\d{2}\.\d{2}\.\d{3}\.\d{4}).*?", result)
+            result_file.write("### guest-kernel-ver :Microsoft Windows [Version %s]\n" % guest_ver[0])
+        else:
+            result_file.write("### guest-kernel-ver :%s" % result)
+    else:
+        result_file.write("### guest-kernel-ver : Microsoft Windows [Version ide driver format]\n")
+
+
+def run(test, params, env):
+    """
+    Block performance test with fio
+    Steps:
+    1) boot up guest with one data disk on specified backend and pin qemu-kvm process to the last numa node on host
+    2) pin guest vcpu and vhost threads to cpus of last numa node repectively
+    3) format data disk and run fio in guest
+    4) collect fio results and host info
+    """
+    def cpu_thread(cpu_cmd, cpufile):
+        cpu_cmd = "%s > %s" % (cpu_cmd, cpufile)
+        commands.getoutput(cpu_cmd)
+
+    def fio_thread(session, run_cmd, cmd_timeout):
+        session.cmd_status(run_cmd, cmd_timeout)
+
+    def _pin_vm_threads(vm, node):
+        if node:
+            if not isinstance(node, utils_misc.NumaNode):
+                node = utils_misc.NumaNode(int(node))
+            utils_test.qemu.pin_vm_threads(vm, node)
+
+        return node
+
+    # login virtual machine
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    login_timeout = int(params.get("login_timeout", 360))
+    session = vm.wait_for_login(timeout=login_timeout)
+    logging.debug(commands.getoutput("numactl --hardware"))
+    logging.debug(commands.getoutput("numactl --show"))
+    _pin_vm_threads(vm, params.get("numa_node"))
+
+    # get parameter from dictionary
+    fio_cmd = params.get('fio_cmd')
+    write_cmd = params.get('write_cmd')
+    rw = params.get("rw")
+    block_size = params.get("block_size")
+    iodepth = params.get("iodepth")
+    threads = params.get("threads")
+    cmd_timeout = int(params.get("cmd_timeout", 1200))
+    order_list = params.get("order_list")
+    driver_format = params.get("drive_format")
+    guest_ver_cmd = params.get("guest_ver_cmd")
+    pattern = params.get("pattern")
+    pre_cmd = params.get("pre_cmd")
+    format = params.get("format")
+    md5value = params.get("md5value")
+    compile_cmd = params.get("compile_cmd")
+    test_src = params.get("test_src")
+    os_type = params.get("os_type")
```

Comment:
No os_type in the cfg, better to set a default value. 
os_type = params.get("os_type", default_value)